app/lib/features/space/widgets/space_sections/space_actions_section.dart

```diff
@@ -37,118 +39,205 @@ class SpaceActionsSection extends ConsumerWidget {
-        simpleActionButton(
-          context: context,
-          iconData: PhosphorIcons.newspaper(),
-          title: lang.addBoost,
-          onPressed: () async {
-            if (!canPostUpdate && canChangeSetting) {
-              final result = await offerToActivateFeature(
-                context: context,
-                ref: ref,
-                spaceId: spaceId,
-                feature: SpaceFeature.boosts,
-              );
-              if (!result) return;
-            }
-            if (context.mounted) {
-              context.pushNamed(
-                Routes.actionAddUpdate.name,
-                queryParameters: {'spaceId': spaceId},
-              );
-            }
-          },
+    return Wrap(
+      crossAxisAlignment: WrapCrossAlignment.start,
+      children: [
+        addPostUpdateButton(
+          context,
+          ref,
+          canChangeSetting,
+          membership,
+          settings,
         ),
-      if (canAddPin || canChangeSetting)
-        simpleActionButton(
-          context: context,
-          iconData: Atlas.pin,
-          title: lang.addPin,
-          onPressed: () async {
-            if (!canAddPin && canChangeSetting) {
-              final result = await offerToActivateFeature(
-                context: context,
-                ref: ref,
-                spaceId: spaceId,
-                feature: SpaceFeature.pins,
-              );
-              if (!result) return;
-            }
-            if (context.mounted) {
-              context.pushNamed(
-                Routes.createPin.name,
-                queryParameters: {'spaceId': spaceId},
-              );
-            }
-          },
+        addPinButton(
+          context,
+          ref,
+          canChangeSetting,
+          membership,
+          settings,
         ),
-      if (canAddEvent || canChangeSetting)
-        simpleActionButton(
-          context: context,
-          iconData: Atlas.calendar_dots,
-          title: lang.addEvent,
-          onPressed: () async {
-            if (!canAddEvent && canChangeSetting) {
-              final result = await offerToActivateFeature(
-                context: context,
-                ref: ref,
-                spaceId: spaceId,
-                feature: SpaceFeature.events,
-              );
-              if (!result) return;
-            }
-            if (context.mounted) {
-              context.pushNamed(
-                Routes.createEvent.name,
-                queryParameters: {'spaceId': spaceId},
-              );
-            }
-          },
+        addEventButton(
+          context,
+          ref,
+          canChangeSetting,
+          membership,
+          settings,
         ),
-      if (canAddTask || canChangeSetting)
-        simpleActionButton(
-          context: context,
-          iconData: Atlas.list,
-          title: lang.addTask,
-          onPressed: () async {
-            if (!canAddEvent && canChangeSetting) {
```

Comment:
_this_ was the actual bug. `canAddEvent` is the wrong check here. However, while looking at it, I also noticed that we are only checking the permission but not if the feature is activated, so I did refactor the entire file to check for the feature-activation more specifically.